common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -1267,6 +1268,119 @@ public static UTF8String[] icuSplitSQL(final UTF8String string, final UTF8String
     return strings.toArray(new UTF8String[0]);
   }
 
+  /**
+   * Title casing a string according to a new behaviour.
+   * Iterates over the string and title cases the first character in each word, and lowercases
+   * every other character.
+   * Handles lowercasing greek sigma(Σ) separately, taking into account if
+   * it should be a small final sigma(ς) or small non-final sigma(σ).
+   * Words are separated by ASCII space(\u0020).
```

Comment:
please make use of the full line length (~100 chars) for comments, without unnecessary newlines

Reply:
(to avoid duplicate comments on the PR - this goes for all instances below)